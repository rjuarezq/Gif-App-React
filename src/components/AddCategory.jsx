import { useState } from "react";
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState('');

    const onInputChange = ({ target }) => {
        setinputValue(target.value);
    }

    const onSubmit = ( e ) => {
        e.preventDefault();
        const newElement = inputValue.trim();
        if(newElement.length <=2) return;
        setinputValue('');
        onNewCategory(newElement);
    }

    return (

        <form onSubmit={ onSubmit }>
            <input 
                className="search" 
                type="text"
                placeholder="Ingrese su bÃºsqueda"
                value = {inputValue}
                onChange={ (e) => onInputChange(e)} >
            </input>
        </form>

    );
};

AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}