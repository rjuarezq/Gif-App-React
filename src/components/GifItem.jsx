import PropTypes from 'prop-types';

export const GifItem = ( {title, url}) => {
    return (
        <div className="card">
            <img src= {url} alt = {title}></img>
            <a target="_blank" href={url} className="card__description"> {title} </a>
        </div>
    );
}

GifItem.propTypes = {
    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
};