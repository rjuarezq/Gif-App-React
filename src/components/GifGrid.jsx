import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const {images, isLoading} = useFetchGifs (category);

    return (
        <div className="category category__container">
                <h3 className="category__title"> {category} </h3>
                {
                    isLoading && ( <h3> Cargando imagenes... </h3> )
                }
                <div className="card__flex">
                    {
                        images.map(( item ) => (
                                <GifItem 
                                    key = { item.id }
                                    //Esparcimos todas props que vienen en item con spread operator
                                    //Tecnica mayormente usada cuando se tiene una gran cantidad de propiedades
                                    {...item}
                                />
                        ))

                    }
                </div>
        </div>
    );
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}
