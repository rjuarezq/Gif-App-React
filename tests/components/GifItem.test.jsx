import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe ('Test in <GifItem.jsx', () => {
    const title = 'Ricardo Juarez';
    const url = 'https://juarezqricardo.com/profile.jpg';
    
    test('It have to match with snapshot', () => {
        const { container } = render(<GifItem title = {title} url = {url}/>);
        expect( container ).toMatchSnapshot();
    });

    test('This test sould show the img with the url and alt attribute', () => {
        render(<GifItem title= {title} url = {url} />);
        //screen.debug(); //show the node element in terminal 
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('This test should show the component title', () => {
        render (<GifItem title = {title} url= {url}/>);
        expect( screen.getByText( title )).toBeTruthy();
    });
});