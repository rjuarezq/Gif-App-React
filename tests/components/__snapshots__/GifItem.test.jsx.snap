// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem.jsx It have to match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Ricardo Juarez"
      src="https://juarezqricardo.com/profile.jpg"
    />
    <a
      class="card__description"
      href="https://juarezqricardo.com/profile.jpg"
      target="_blank"
    >
       
      Ricardo Juarez
       
    </a>
  </div>
</div>
`;
